File: CombatTracker/CombatTrackerData/AttacksAddCrit.vb
<Bind(Include:="Critical")> _
Partial Class AttacksAddCrit
    Inherits IEntity(Of AttacksAddCrit)

    Private Sub OnCreated()
        'Critical = CritTypes.None
    End Sub


    Private Sub OnValidate(ByVal action As System.Data.Linq.ChangeAction)
        If action <> Data.Linq.ChangeAction.Delete AndAlso Not IsValid Then
            Throw New ApplicationException("Rule violations prevent saving")
        End If
    End Sub

    Public Overrides Function GetRuleViolations() As System.Collections.Generic.List(Of RuleViolation)
        Dim L As New List(Of RuleViolation)
        'If String.IsNullOrEmpty(Name) Then
        '    L.Add(New RuleViolation("Name Required", "Name"))
        'End If
        If Attack Is Nothing Then
            L.Add(New RuleViolation("Must be on an attack", "AttackID"))
        End If
        If CriticalType Is Nothing AndAlso Critical_ID = 0 Then
            L.Add(New RuleViolation("Critical must be set.", "Critical_ID"))
        End If
        Return L
    End Function
End Class

File: CombatTracker/Combat.Domain/AttacksAddCrit.cs
namespace Combat.Domain
{
    using System;
    using System.Collections.Generic;
    using System.ComponentModel.DataAnnotations;
    using System.ComponentModel.DataAnnotations.Schema;
    using System.Data.Entity.Spatial;

    public partial class AttacksAddCrit
    {
        public int ID { get; set; }

        public int? AttackID { get; set; }

        public int Critical_ID { get; set; }

        public virtual Attack_CriticalTypes Attack_CriticalTypes { get; set; }

        public virtual Attack Attack { get; set; }
    }
}
